### PR TITLE
feat: text classification for benchmarks

### PR DESCRIPTION
**Pull request by aski02**
### Changes (#265)
1. Refactored Kronfluence wrapper to handle text classification correctly (+bug fixes).
3. Added dataset handler class for handling both torch and hugging face datasets.
4. Modified `benchmarks/base.py` to handle inference on hugging face datasets correctly.
5. Corrected `class_detection` and `top_k_cardinality` benchmark and metric class to support text classification.
6. Added `slow` marker for computationally expensive tests (will not be run automatically).
7. Added dummy text classification dataset and model to `conftest.py`.
8. Added small text classification test for `kronfluence.py`.
9. Added both mnist and text classification tests for `class_detection` and `top_k_cardinality` benchmarks using Kronfluence.

### CODE REVIEW

File: quanda/metrics/downstream_eval/class_detection.py [hunk 1]
```diff
@@ -61,6 +62,36 @@ def __init__(
 
         self.scores: List[torch.Tensor] = []
 
+    def _get_targets(self, item: Union[tuple, dict]):
```
Comment (dilyabareeva) [inline quanda/metrics/downstream_eval/class_detection.py]: Could we move this to the base metric class?

File: quanda/metrics/heuristics/mixed_datasets.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 """Mixed Datasets Metric."""
 
-from typing import Any, List, Optional, Union, Callable
+from typing import Any, Callable, List, Optional, Union
```
Comment (dilyabareeva) [inline quanda/metrics/heuristics/mixed_datasets.py:3]: Which formatter do you use? The one we have in the Makefile under `clean-format` doesn't do import sorting, but it would be nice if it did. Is it a `ruff` setting?

File: quanda/utils/common.py [hunk 3]
```diff
@@ -313,6 +311,30 @@ def process_targets(
     return targets
 
 
+def move_to_device(
```
Comment (dilyabareeva) [inline quanda/utils/common.py]: Maybe let's call it `move_ds_item_to_device` or something like that, more descriptive?

File: quanda/metrics/downstream_eval/class_detection.py (hunk 1 above)
Comment (dilyabareeva) [reply quanda/metrics/downstream_eval/class_detection.py]: Or maybe do the dataset handler?

File: quanda/utils/datasets/dataset_handlers.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        Returns
+        -------
+        Any
+            Model-ready inputs.
+
+        """
+        pass
+
+    @abstractmethod
+    def extract_predictions(self, outputs: Any) -> torch.Tensor:
```
Comment (dilyabareeva) [inline quanda/utils/datasets/dataset_handlers.py]: `get_predictions`?

File: quanda/utils/datasets/dataset_handlers.py [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+
+class TorchDatasetHandler(DatasetHandler):
+    """Handler for PyTorch datasets."""
+
+    def process_batch(
+        self,
+        batch: Tuple[torch.Tensor, torch.Tensor],
+        device: Union[str, torch.device],
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Process a batch of data.
```
Comment (dilyabareeva) [inline quanda/utils/datasets/dataset_handlers.py:106]: How is this different from `move_to_device` in common utils?

File: quanda/utils/datasets/dataset_handlers.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+        A handler instance suited for the dataset.
+
+    Raises
+    ------
+    ValueError
+        If the dataset is not supported.
+
+    """
+    if isinstance(dataset, datasets.Dataset):
+        return HuggingFaceDatasetHandler()
```
Comment (dilyabareeva) [inline quanda/utils/datasets/dataset_handlers.py:294]: Here we could already check if a dataset item contains "labels" key, and throw an error if it doesn't (for some early prevention)

File: tests/conftest.py [hunk 7]
```diff
@@ -852,3 +853,58 @@ def load_wandb_config():
 def load_tensorboard_config():
     """Load TensorBoard config from tensorboard.yaml as a Python dict."""
     return load_yaml("config/logger/tensorboard.yaml")
+
+
+@pytest.fixture
+def load_simple_classifier():
+    torch.manual_seed(42)
+
+    class SimpleTextClassifier(nn.Module):
```
Comment (dilyabareeva) [inline tests/conftest.py]: can we move this to `tests/models.py`?

**Review by dilyabareeva — COMMENTED** [review]
Hi @aski02, thanks a lot for the great work! I have only very small suggestions. I think it also would be great if you could apply the dataset handler to all the metrics, not just `class_detection`. Then I think we are good to merge :)

File: quanda/utils/datasets/dataset_handlers.py (hunk 5 above)
Comment (aski02) [reply quanda/utils/datasets/dataset_handlers.py:106]: It's also used to split the batch into inputs and labels (which is handled differently for pytorch datasets vs huggingface datasets.

**Comment by aski02** [comment]
> Hi @aski02, thanks a lot for the great work! I have only very small suggestions. I think it also would be great if you could apply the dataset handler to all the metrics, not just `class_detection`. Then I think we are good to merge :)

Thanks for the feedback! We currently use the dataset handler only in the benchmark base class when computing predictions. All the other suggestions you've made are fixed and pushed now.
